ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -13900,7 +13901,11 @@ private void addDbAndTabToOutputs(String[] qualifiedTabName, TableType type,
     for(Map.Entry<String,String> serdeMap : storageFormat.getSerdeProps().entrySet()){
       t.setSerdeParam(serdeMap.getKey(), serdeMap.getValue());
     }
-    outputs.add(new WriteEntity(t, WriteEntity.WriteType.DDL_NO_LOCK));
+    if (tblProps.get("created_with_ctas") == "true") {
+      outputs.add(new WriteEntity(t, WriteType.CTAS));
```

Comment:
Why is this needed/when, we are populating outputs in CreateTableOperation as well?

Reply:
In CreateTableOperation, DDLUtils.addIfAbsentByName would populate the output only if it was missed in the SemanticAnalyzer stage. 

So I think, removing this step in CreateTableOperation and retaining it only in SemanticAnalyze would be fine.